### PR TITLE
Starting the HighAvailabilityServices

### PR DESCRIPTION
**Pull request by sundargates**
### Context

Explain context and other details for this pull request.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-client/src/main/java/io/mantisrx/client/MantisClient.java [hunk 1]
```diff
@@ -101,6 +102,9 @@ public MantisClient(Properties properties) {
         HighAvailabilityServices haServices =
             HighAvailabilityServicesUtil.createHAServices(
                 Configurations.frmProperties(properties, CoreConfiguration.class));
+        if (haServices.state() == State.NEW) {
+            haServices.startAsync().awaitRunning();
```
Comment (Andyz26) [inline mantis-client/src/main/java/io/mantisrx/client/MantisClient.java:106]: Is it possible for the haServices not being started right now but started later?

Comment (sundargates) [reply mantis-client/src/main/java/io/mantisrx/client/MantisClient.java:106]: In that case, you should use the other constructor that allows you to control the lifecycle of the HighAvailabilityServices better.

File: mantis-common/src/main/java/com/mantisrx/common/utils/Services.java [hunk 2]
```diff
@@ -21,20 +21,23 @@
...
+     * to start, then the future is completed exceptionally with the Throwable cause.
      *
-     * @param service service that needs to be started
+     * @param service  service that needs to be started
      * @param executor executor to be used for notifying the caller.
      * @return a future
      */
     public static CompletableFuture<Void> startAsync(Service service, Executor executor) {
-        Preconditions.checkArgument(service.state() == State.NEW, "Assumes the service has not been started yet");
+        Preconditions.checkArgument(service.state() == State.NEW,
```
Comment (Andyz26) [inline mantis-common/src/main/java/com/mantisrx/common/utils/Services.java:39]: is this call thread-safe? is lock needed?

File: mantis-common/src/main/java/com/mantisrx/common/utils/Services.java [hunk 3]
```diff
@@ -89,4 +92,19 @@ public static CompletableFuture<Void> stopAsync(Service service, Executor execut
...
+    /**
+     * Attempts to start the service and waits for it to reach the RUNNING state. If the service has
+     * already started, then this method returns immediately.
+     *
+     * @param service
+     */
+    public static void startAndWait(Service service) {
+        try {
+            service.startAsync();
+        } catch (IllegalStateException e) {
```
Comment (Andyz26) [inline mantis-common/src/main/java/com/mantisrx/common/utils/Services.java:105]: check service state instead of try/catch?

**Review by Andyz26 — APPROVED** [review]

File: mantis-common/src/main/java/com/mantisrx/common/utils/Services.java (hunk 2 above)
Comment (sundargates) [reply mantis-common/src/main/java/com/mantisrx/common/utils/Services.java:39]: This is the existing code. Not related to my changes.

**Pull request merged**